lib/ansible/modules/cloud/openstack/os_port.py

```diff
@@ -228,21 +244,35 @@ def _needs_update(module, port, cloud):
                       'device_id',
                       'binding:vnic_type',
                       'port_security_enabled']
-    compare_dict = ['allowed_address_pairs',
-                    'extra_dhcp_opts']
+    compare_list_dict = ['allowed_address_pairs',
+                         'extra_dhcp_opts']
     compare_list = ['security_groups']
 
     for key in compare_simple:
         if module.params[key] is not None and module.params[key] != port[key]:
             return True
-    for key in compare_dict:
-        if module.params[key] is not None and module.params[key] != port[key]:
-            return True
     for key in compare_list:
         if module.params[key] is not None and (set(module.params[key]) !=
                                                set(port[key])):
             return True
 
+    for key in compare_list_dict:
+        if not module.params[key]:
+            if not port[key]:
+                return True
+
+            # sort dicts in list
+            port_ordered = [OrderedDict(sorted(d.items())) for d in port[key]]
```

Comment:
Just an idea. Wouldn't you get the same result if you simply have
```suggestion
            port_ordered = [sorted(d.items()) for d in port[key]]
```
? `OrderedDict(sorted(a.items())) == OrderedDict(sorted(b.items()))` should be true if and only if `sorted(a.items()) == sorted(b.items())` (for `dict`s `a` and `b`).

Reply:
It's been a while since I submitted this PR, afraid I can't really remember my reasoning behind current solution, I may have been determined to compare dicts not realizing I've could've made it simpler :). However, most openstack modules require python >= 2.7 where ordereddict is included anyway, requiring py2.7 may not be a breaking change for many

Reply:
Fine for me, as long as the maintainers agree :)